src/lib/HST/Frontend/Syntax.hs

```diff
@@ -78,9 +78,9 @@ deriving instance ShowAST a => Show (Decl a)
 
 -- | A data constructor. Does not include a source span and should not be
 --   transformed back.
-data ConDecl a = ConDecl (Name a) [TypeExp a]
-               | InfixConDecl (TypeExp a) (Name a) (TypeExp a)
-deriving instance EqAST a => Eq (ConDecl a)
+data ConDecl a = ConDecl (Name a) Int
+               | InfixConDecl (Name a)
```

Comment:
Only the `ConDecl` constructor and a `isInfix` flag would probably suffice, too.

Reply:
Do you want a new type for this flag or a `Bool`?
Either way, I don't think two constructors for `ConDecl` or a flag make much of a difference, but I will change it if you want.

Reply:
> I don't think two constructors for `ConDecl` or a flag make much of a difference

I agree. The only difference it would make is that we could reduce some code duplication in `HST.Application`.

> Do you want a new type for this flag or a Bool?

I'd suggest something like this.

```suggestion
data ConDecl a
  = ConDecl { conDeclName    :: Name a
            , conDeclArity   :: Int
            , conDeclIsInfix ::  Bool
            }
```